### DISCUSSION TITLE
Top keywords comparison with job description

### DISCUSSION THREAD

**Discussion by frankhart2018** [Documentive/TemplateBuddy#97] [Ideas]
**Problem**

Most companies use automated keyword-based filtering to perform a first pass over candidate resumes and filter them out based on how many keywords from their job description match with the submitted resume, if it is below a certain threshold, then the 
the candidate is sent a rejection email/message. 

**Proposed Solution**

A keyword matcher between the job description and the completely filled resume of a candidate, assessing which keywords are missing in the candidate's resume when compared against a job description that the person provided us with. 

**Initial Implementation Ideas**

We generate a word cloud and extract top-K keywords from the description provided, we do the same for the content of a person's resume (by filtering out all the topics/headers and only taking the relevant content into consideration). Once we do so, we get top-K keywords in both texts and then match them against each other to find out the difference between top-K from the job description and top-K from the resume).

**Comment by dhairyaj** [comment]
Seems like a good idea to me. This requires a separate option/UI for taking in the resume and providing a comparison between the resume and JD to find how appropriate the candidate's resume is for the proposed JD
- Could be a possible extension of the TemplateBuddy utility.

### Proposed Implementation
We can add the option in the dashboard after the user logs in. User can select the option to compare and in the UI, JD can be provided and the user can select which resume needs to be compared and accordingly the output is given.
- Access to any of resume is provided as in TemplateBuddy there would be all the resumes which were either created by manually adding details or generated after importing a JSON Schema.
